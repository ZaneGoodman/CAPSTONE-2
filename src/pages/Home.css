.Home-container {
  margin: auto;
  margin-top: 20px;
  text-align: center;
  align-items: center;
  width: 50rem;
}

h1,
button,
small,
h2,
span,
h3 {
  font-family: "Rosarivo", cursive;
}

.Home-span {
  margin: 100px;
}
.Home-span h3 {
  border-color: rgb(151, 145, 145);
  border-width: 3px;
  border-style: inset;
  background-color: rgb(238, 195, 162);
  width: 50%;
  margin: auto;
}
.Home-span p {
  font-size: 18px;
  font-weight: 600;
  margin-top: 5px;
}
.Home-date button {
  margin: 10px;
}

.Home-span small {
  color: rgb(240, 11, 11);
}

#Home-page {
  background-image: url("https://images.freeimages.com/images/large-previews/954/wooden-rosary-1188070.jpg");
  background-size: cover;
  background-repeat: no-repeat;
  position: absolute;
  margin: 0;
  padding: 0;
  left: 0;
  top: 49px;
  bottom: 0;
  right: 0;
  overflow: scroll;
}

.Home-candle-left img {
  float: left;
  width: 20rem;
  position: relative;
  animation: forwards;
  border-radius: 5px;
  margin-left: 5px;
  margin-top: 5px;
}
.Home-candle-right img {
  float: right;
  width: 20rem;
  position: relative;
  border-radius: 5px;
  margin-right: 5px;
  margin-top: 5px;
}

.Home-calendar {
  position: relative;
  margin: auto;
  margin-bottom: 15px;
}
