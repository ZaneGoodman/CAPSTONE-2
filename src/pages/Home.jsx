import React, { useEffect, useState } from "react";
import Calendar from "react-calendar";
import formatDate from "../helpers/formatDate";
import NotesForm from "../forms/NotesForm";
import "react-dropdown/style.css";
import { useAuth } from "../provider/authProvider";
import Rosary from "../models/rosary";
import prayers from "../models/staticRosaryData";
import Notes from "../models/notes";

const Home = () => {
  const { username } = useAuth();
  const [ros, setRos] = useState({});
  const [season, setSeason] = useState();
  const [date, setDate] = useState(new Date());
  const [showCalendar, setShowCalendar] = useState(false);
  const [addedNotes, setAddedNotes] = useState(false);

  const currDate = String(new Date()).slice(0, 15);
  const pickedDay = String(date).slice(0, 15);

  useEffect(() => {
    const getRosary = async () => {
      //Get Rosary info based on date, add data to state
      const rosaryInfo = await Rosary.getByDate(formatDate(date));
      setRos((r) => ({ ...r, ...rosaryInfo.data[0] }));
      setSeason(rosaryInfo.season);
      //Track if notes have been added,add to state hide form if they have
      const didAddNotes = await Notes.checkDateExists(
        username,
        formatDate(new Date())
      );
      setAddedNotes(didAddNotes);
    };

    getRosary();
  }, [date, addedNotes, username]);

  const toggleCalendar = () =>
    //Toggle showing calendar based on button click
    showCalendar === true ? setShowCalendar(false) : setShowCalendar(true);

  const parseAnnounces = () => {
    let annouceQueue = [];
    for (let i = 1; i <= 5; i++) {
      const announce = (
        <div key={i}>
          <span>
            <h3>Announce {i}</h3>
            <p>{ros[`announce_${i}`]}</p>
          </span>
          <span>
            <h3>Our Father</h3>
            <p>{prayers.getOurFather()}</p>
          </span>
          <span>
            <h3>Hail Mary X10</h3>
            <small>*While meditating on the rosary mystery*</small>
            <p>{prayers.getHailMary()}</p>
          </span>
          <span>
            <h3>Glory Be to the Father</h3>
            <small>*Holding Chain*</small>
            <p>{prayers.getGloryBe()}</p>
          </span>
          <span>
            <h3>Fatima Prayer (oh my Jesus)</h3>
            <p>{prayers.getFatima()}</p>
          </span>
        </div>
      );
      annouceQueue.push(announce);
    }
    return annouceQueue;
  };

  return (
    <div>
      {showCalendar && (
        <Calendar data-testid="calendar" onChange={setDate} value={date} />
      )}
      <h1>{season}</h1>
      <small>
        {pickedDay}
        <button onClick={toggleCalendar}>Pick Date</button>
      </small>
      <h2>The {ros.group_by} Mystery</h2>
      <span>
        <h3>Apostles Creeed</h3>
        <small>*Holding Crucifix*</small>
        <p>{prayers.getApostlesCreed()}</p>
      </span>
      <span>
        <h3>Our Father</h3>
        <p>{prayers.getOurFather()}</p>
      </span>
      <span>
        <h3>Hail Mary X3</h3>
        <p>{prayers.getHailMary()}</p>
      </span>
      <span>
        <h3>Glory Be to the Father</h3>
        <small>*Holding Chain*</small>
        <p>{prayers.getGloryBe()}</p>
      </span>
      {parseAnnounces()}
      <span>
        <h3>Hail, Holy Queen</h3>
        <p>{prayers.getHailHolyQueen()}</p>
      </span>
      <span>
        <h3>Final Prayer</h3>
        <p>{prayers.getFinalPrayer()}</p>
      </span>
      <span>
        <h3>{prayers.getInTheName()}</h3>
      </span>
      {currDate === pickedDay && addedNotes === false && (
        <NotesForm
          date={formatDate(date)}
          season={season}
          username={username}
          setAddedNotes={setAddedNotes}
        />
      )}
    </div>
  );
};

export default Home;
