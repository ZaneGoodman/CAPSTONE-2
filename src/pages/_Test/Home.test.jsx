import React from "react";
import { render, fireEvent } from "@testing-library/react";

import AuthProvider from "../../provider/authProvider";
import Home from "../Home";
//Smoke test
test("renders Home component with authentication", async () => {
  render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <Home />
    </AuthProvider>
  );
});

//Snapshot test
it("Card matches snapshot", async () => {
  const { asFragment } = render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <Home />
    </AuthProvider>
  );
  await expect(asFragment()).toMatchSnapshot();
});

test("test calendar visibility", () => {
  const { getByText } = render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <Home />
    </AuthProvider>
  );

  const pickDateButton = getByText("Pick Date");
  fireEvent.click(pickDateButton);
  expect(getByText("Fri")).toBeInTheDocument();
  fireEvent.click(pickDateButton);
});

test("Test dynamic announces", () => {
  const { getByText } = render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <Home />
    </AuthProvider>
  );
  const thing = getByText("Hail, Holy Queen");
  expect(thing).toBeInTheDocument();
});
