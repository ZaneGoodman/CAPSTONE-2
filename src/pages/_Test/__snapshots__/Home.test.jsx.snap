// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1 />
    <small>
      Tue Dec 05 2023
      <button>
        Pick Date
      </button>
    </small>
    <h2>
      The  Mystery
    </h2>
    <span>
      <h3>
        Apostles Creeed
      </h3>
      <small>
        *Holding Crucifix*
      </small>
      <p>
        I believe in God, the Father almighty, Creator of heaven and earth, and in Jesus Christ, his only Son, our Lord, who was conceived by the Holy Spirit, born of the Virgin Mary, suffered under Pontius Pilate, was crucified, died and was buried; he descended into hell; on the third day he rose again from the dead;he ascended into heaven, and is seated at the right hand of God the Father almighty; from there he will come to judge the living and the dead. I believe in the Holy Spirit, the holy catholic Church, the communion of saints, the forgiveness of sins, the resurrection of the body, and life everlasting. Amen.
      </p>
    </span>
    <span>
      <h3>
        Our Father
      </h3>
      <p>
        Our Father, who art in heaven, hallowed be thy name; thy kingdom come; thy will be done on earth as it is in heaven. Give us this day our daily bread; and forgive us our trespasses as we forgive those who trespass against us; and lead us not into temptation, but deliver us from evil. Amen.
      </p>
    </span>
    <span>
      <h3>
        Hail Mary X3
      </h3>
      <p>
        Hail Mary, full of grace, the Lord is with you; blessed are you among women, and blessed is the fruit of your womb, Jesus. Holy Mary, Mother of God, pray for us sinners now and at the hour of our death. Amen.
      </p>
    </span>
    <span>
      <h3>
        Glory Be to the Father
      </h3>
      <small>
        *Holding Chain*
      </small>
      <p>
        Glory be to the Father, the Son, and the Holy Spirit; as it was in the beginning, is now, and ever shall be, world without end. Amen.
      </p>
    </span>
    <div>
      <span>
        <h3>
          Announce 1
        </h3>
        <p />
      </span>
      <span>
        <h3>
          Our Father
        </h3>
        <p>
          Our Father, who art in heaven, hallowed be thy name; thy kingdom come; thy will be done on earth as it is in heaven. Give us this day our daily bread; and forgive us our trespasses as we forgive those who trespass against us; and lead us not into temptation, but deliver us from evil. Amen.
        </p>
      </span>
      <span>
        <h3>
          Hail Mary X10
        </h3>
        <small>
          *While meditating on the rosary mystery*
        </small>
        <p>
          Hail Mary, full of grace, the Lord is with you; blessed are you among women, and blessed is the fruit of your womb, Jesus. Holy Mary, Mother of God, pray for us sinners now and at the hour of our death. Amen.
        </p>
      </span>
      <span>
        <h3>
          Glory Be to the Father
        </h3>
        <small>
          *Holding Chain*
        </small>
        <p>
          Glory be to the Father, the Son, and the Holy Spirit; as it was in the beginning, is now, and ever shall be, world without end. Amen.
        </p>
      </span>
      <span>
        <h3>
          Fatima Prayer (oh my Jesus)
        </h3>
        <p>
          O my Jesus, forgive us our sins. Save us from the fires of hell. Lead all souls into heaven, especially those in most need of thy mercy. Amen.
        </p>
      </span>
    </div>
    <div>
      <span>
        <h3>
          Announce 2
        </h3>
        <p />
      </span>
      <span>
        <h3>
          Our Father
        </h3>
        <p>
          Our Father, who art in heaven, hallowed be thy name; thy kingdom come; thy will be done on earth as it is in heaven. Give us this day our daily bread; and forgive us our trespasses as we forgive those who trespass against us; and lead us not into temptation, but deliver us from evil. Amen.
        </p>
      </span>
      <span>
        <h3>
          Hail Mary X10
        </h3>
        <small>
          *While meditating on the rosary mystery*
        </small>
        <p>
          Hail Mary, full of grace, the Lord is with you; blessed are you among women, and blessed is the fruit of your womb, Jesus. Holy Mary, Mother of God, pray for us sinners now and at the hour of our death. Amen.
        </p>
      </span>
      <span>
        <h3>
          Glory Be to the Father
        </h3>
        <small>
          *Holding Chain*
        </small>
        <p>
          Glory be to the Father, the Son, and the Holy Spirit; as it was in the beginning, is now, and ever shall be, world without end. Amen.
        </p>
      </span>
      <span>
        <h3>
          Fatima Prayer (oh my Jesus)
        </h3>
        <p>
          O my Jesus, forgive us our sins. Save us from the fires of hell. Lead all souls into heaven, especially those in most need of thy mercy. Amen.
        </p>
      </span>
    </div>
    <div>
      <span>
        <h3>
          Announce 3
        </h3>
        <p />
      </span>
      <span>
        <h3>
          Our Father
        </h3>
        <p>
          Our Father, who art in heaven, hallowed be thy name; thy kingdom come; thy will be done on earth as it is in heaven. Give us this day our daily bread; and forgive us our trespasses as we forgive those who trespass against us; and lead us not into temptation, but deliver us from evil. Amen.
        </p>
      </span>
      <span>
        <h3>
          Hail Mary X10
        </h3>
        <small>
          *While meditating on the rosary mystery*
        </small>
        <p>
          Hail Mary, full of grace, the Lord is with you; blessed are you among women, and blessed is the fruit of your womb, Jesus. Holy Mary, Mother of God, pray for us sinners now and at the hour of our death. Amen.
        </p>
      </span>
      <span>
        <h3>
          Glory Be to the Father
        </h3>
        <small>
          *Holding Chain*
        </small>
        <p>
          Glory be to the Father, the Son, and the Holy Spirit; as it was in the beginning, is now, and ever shall be, world without end. Amen.
        </p>
      </span>
      <span>
        <h3>
          Fatima Prayer (oh my Jesus)
        </h3>
        <p>
          O my Jesus, forgive us our sins. Save us from the fires of hell. Lead all souls into heaven, especially those in most need of thy mercy. Amen.
        </p>
      </span>
    </div>
    <div>
      <span>
        <h3>
          Announce 4
        </h3>
        <p />
      </span>
      <span>
        <h3>
          Our Father
        </h3>
        <p>
          Our Father, who art in heaven, hallowed be thy name; thy kingdom come; thy will be done on earth as it is in heaven. Give us this day our daily bread; and forgive us our trespasses as we forgive those who trespass against us; and lead us not into temptation, but deliver us from evil. Amen.
        </p>
      </span>
      <span>
        <h3>
          Hail Mary X10
        </h3>
        <small>
          *While meditating on the rosary mystery*
        </small>
        <p>
          Hail Mary, full of grace, the Lord is with you; blessed are you among women, and blessed is the fruit of your womb, Jesus. Holy Mary, Mother of God, pray for us sinners now and at the hour of our death. Amen.
        </p>
      </span>
      <span>
        <h3>
          Glory Be to the Father
        </h3>
        <small>
          *Holding Chain*
        </small>
        <p>
          Glory be to the Father, the Son, and the Holy Spirit; as it was in the beginning, is now, and ever shall be, world without end. Amen.
        </p>
      </span>
      <span>
        <h3>
          Fatima Prayer (oh my Jesus)
        </h3>
        <p>
          O my Jesus, forgive us our sins. Save us from the fires of hell. Lead all souls into heaven, especially those in most need of thy mercy. Amen.
        </p>
      </span>
    </div>
    <div>
      <span>
        <h3>
          Announce 5
        </h3>
        <p />
      </span>
      <span>
        <h3>
          Our Father
        </h3>
        <p>
          Our Father, who art in heaven, hallowed be thy name; thy kingdom come; thy will be done on earth as it is in heaven. Give us this day our daily bread; and forgive us our trespasses as we forgive those who trespass against us; and lead us not into temptation, but deliver us from evil. Amen.
        </p>
      </span>
      <span>
        <h3>
          Hail Mary X10
        </h3>
        <small>
          *While meditating on the rosary mystery*
        </small>
        <p>
          Hail Mary, full of grace, the Lord is with you; blessed are you among women, and blessed is the fruit of your womb, Jesus. Holy Mary, Mother of God, pray for us sinners now and at the hour of our death. Amen.
        </p>
      </span>
      <span>
        <h3>
          Glory Be to the Father
        </h3>
        <small>
          *Holding Chain*
        </small>
        <p>
          Glory be to the Father, the Son, and the Holy Spirit; as it was in the beginning, is now, and ever shall be, world without end. Amen.
        </p>
      </span>
      <span>
        <h3>
          Fatima Prayer (oh my Jesus)
        </h3>
        <p>
          O my Jesus, forgive us our sins. Save us from the fires of hell. Lead all souls into heaven, especially those in most need of thy mercy. Amen.
        </p>
      </span>
    </div>
    <span>
      <h3>
        Hail, Holy Queen
      </h3>
      <p>
        Hail, holy Queen, mother of mercy, our life, our sweetness, and our hope. To you we cry, poor banished children of Eve; to you we send up our sighs, mourning and weeping in this valley of tears. Turn, then, most gracious advocate, your eyes of mercy toward us; and after this, our exile, show unto us the blessed fruit of your womb, Jesus. O clement, O loving, O sweet Virgin Mary.
      </p>
    </span>
    <span>
      <h3>
        Final Prayer
      </h3>
      <p>
        O God, whose only begotten Son, by His life, death, and resurrection, has purchased for us the rewards of eternal life, grant, we beseech Thee, that meditating upon these mysteries of the Most Holy Rosary of the Blessed Virgin Mary, we may imitate what they contain and obtain what they promise, through the same Christ Our Lord. Amen.
      </p>
    </span>
    <span>
      <h3>
        In the name of the Father, and of the Son, and of the Holy Spirit. Amen.
      </h3>
    </span>
    <form>
      <label
        for="notes"
      >
        Notes
      </label>
      <textarea
        cols="100"
        id="notes"
        name="notes"
        rows="15"
      />
      <label
        for="marker"
      >
        Prayed today?
      </label>
      <input
        id="marker"
        name="marker"
        type="checkbox"
        value="false"
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
